pkg/build/api/types.go

```diff
@@ -185,6 +185,11 @@ type CustomBuildStrategy struct {
 	// a validation error is returned.
 	Image string `json:"image"`
 
+	// DockerRegistryAuthRef is used to provide Docker Registry authentication
+	// secret that is used for communication with authentication enabled Docker
+	// Registry. This have to reference existing Secret object.
+	DockerRegistrySecretRef *kapi.ObjectReference `json:"dockerRegistrySecretRef,omitempty"`
```

Comment:
I see it does, so that question stands.  i kind of like keeping it at the same level as the image it's associated with, but i hate the fact that we have the same field 3 times.